tests/acceptance/helpers/httpHelper.js

```diff
@@ -49,3 +63,204 @@ exports.checkOCSStatus = function (response, message) {
+ * @param {object} params
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.delete = function (url, params, userId, header) {
+  const options = { ...params, method: 'DELETE' }
+  return this.requestEndpoint(url, options, userId, header)
+}
+
+/**
+ *
+ * @param {string} url
+ * @param {object} params
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.move = function (url, params, userId, header) {
+  const options = { ...params, method: 'MOVE' }
+  return this.requestEndpoint(url, options, userId, header)
+}
+
+/**
+ *
+ * @param {string} url
+ * @param {object} params
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.propfind = function (url, params, userId, header) {
+  const options = { ...params, method: 'PROPFIND' }
+  return this.requestEndpoint(url, options, userId, header)
+}
+
+/**
+ *
+ * @param {string} url
+ * @param {object} params
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.report = function (url, params, userId, header) {
+  const options = { ...params, method: 'REPORT' }
+  return this.requestEndpoint(url, options, userId, header)
+}
+
+/**
+ *
+ * @param {string} url
+ * @param {object} params
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.put = function (url, params, userId, header) {
+  const options = { ...params, method: 'PUT' }
+  return this.requestEndpoint(url, options, userId, header)
+}
+
+/**
+ *
+ * @param {string} url
+ * @param {object} params
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.proppatch = function (url, params, userId, header) {
+  const options = { ...params, method: 'PROPPATCH' }
+  return this.requestEndpoint(url, options, userId, header)
+}
+
+/**
+ *
+ * @param {string} path
+ * @param {object} params
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.requestOCSEndpoint = function (path, params, userId = 'admin', header = {}) {
+  const headers = { ...this.createOCSRequestHeaders(userId), ...header }
+  const options = { ...params, headers }
+  const separator = path.includes('?') ? '&' : '?'
+  const url = join(backendHelper.getCurrentBackendUrl(), 'ocs/v2.php', path + separator + 'format=json')
+  return fetch(url, options)
+}
+
+/**
+ *
+ * @param {string} url
+ * @param {object} params
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.getOCS = function (url, params, userId, header) {
+  const options = { ...params, method: 'GET' }
+  return this.requestOCSEndpoint(url, options, userId, header)
+}
+
+/**
+ *
+ * @param {string} url
+ * @param {object} params
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.postOCS = function (url, params, userId, header) {
```

Comment:
can we do all these functions like
```js
module.exports = {
 deleteOCS: (url, params, userId, header) => this.requestOCSEndpoint( url, {...params, method: 'DELETE'}, userId, header),
 getOCS: (url, params, userId, header) => this.requestOCSEndpoint( url, {...params, method: 'GET'}, userId, header),
 putOCS: (url, params, userId, header) => this.requestOCSEndpoint( url, {...params, method: 'PUT'}, userId, header)
}
```
I think that makes it easier to follow and I dont think we really need the docstring for each of these functions

Reply:
@dpakach I have been thinking about that, but right now I am trying to make CI green